Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -1761,6 +1761,13 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         
         try shouldContinueRegistration()
         
+        // Articles that will be automatically curated can be resolved but they need to be pre registered before resolving links.
+        let rootNodeForAutomaticCuration = rootModules.count == 1 ? rootModules.first.flatMap(topicGraph.nodeWithReference) : nil
+        if rootNodeForAutomaticCuration != nil {
+            registerArticles(otherArticles, in: bundle)
```

Comment:
I'm aware that the order that task are performed in this method does matter but I believe that registering articles earlier is a safe change to make.